server/src/main/java/io/crate/replication/logical/LogicalReplicationService.java

```diff
@@ -79,4 +185,292 @@ public void clusterChanged(ClusterChangedEvent event) {
+
+                @Override
+                public void onResponse(PublicationsStateAction.Response stateResponse) {
+                    verifyTablesDoNotExist(
+                        subscriptionName,
+                        stateResponse,
+                        r -> {
+                            for (var index : stateResponse.concreteIndices()) {
+                                subscribedIndices.put(index, subscriptionName);
+                            }
+                            initiateReplication(subscriptionName, subscription, stateResponse, listener);
+                        },
+                        listener::onFailure
+                    );
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    listener.onFailure(e);
+                }
+            }
+        );
+    }
+
+    public void verifyTablesDoNotExist(String subscriptionName,
+                                       Subscription subscription,
+                                       Consumer<Void> onSuccess,
+                                       Consumer<Exception> onFailure) {
+        getPublicationState(
+            subscriptionName,
+            subscription,
+            ActionListener.wrap(x -> verifyTablesDoNotExist(subscriptionName, x, onSuccess, onFailure), onFailure)
+        );
+    }
+
+    private void getPublicationState(String subscriptionName, Subscription subscription, ActionListener<PublicationsStateAction.Response> listener) {
+        updateRemoteConnection(
+            subscriptionName,
+            subscription.connectionInfo().settings(),
+            ActionListener.delegateFailure(
+                listener,
+                (delegatedListener, response) -> {
+                    var remoteClient = getRemoteClusterClient(
+                        threadPool,
+                        subscriptionName
+                    );
+                    remoteClient.execute(
+                        PublicationsStateAction.INSTANCE,
+                        new PublicationsStateAction.Request(subscription.publications()),
+                        listener
+                    );
+                }
+            )
+        );
+    }
+
+    private void verifyTablesDoNotExist(String subscriptionName,
+                                        PublicationsStateAction.Response stateResponse,
+                                        Consumer<Void> onSuccess,
+                                        Consumer<Exception> onFailure) {
+        var metadata = clusterService.state().metadata();
+        Consumer<RelationName> onExists = (relation) -> {
+            var message = String.format(
+                Locale.ENGLISH,
+                "Subscription '%s' cannot be created as included relation '%s' already exists",
+                subscriptionName,
+                relation
+            );
+            onFailure.accept(new RelationAlreadyExists(relation, message));
```

Comment:
From a UX point of view I think it is not ideal that the user will never see this error unless they're reading the logs. Will this pop up in the sys/pg_catalog tables we're planning to add?

Reply:
Currently we check for existing tables twice, once on subscription creation (errror bubbles up to the user) and once when the replication starts. As the replications is triggered by cluster state changes now, a error can only be logged and maybe exposed via some system tables. But the PG definition of `pg_subscription` does not define a column for errors, see https://www.postgresql.org/docs/current/catalog-pg-subscription.html. (I guess the initial replication at PG is trigggered by the create subscription command and thus an error will be bubbled up).
But on subscriptions on FOR ALL TABLES publications, an error can always happen anyway so I think we should really expose errors somewhere in system tables.

That's the drawback of triggering the replication asynchronously via the cluster state listener. I'd tend to revert that change as it also causes a bit of friction, like the create subscription command returns while initial replication is still ongoing (we could of course think of a `wait_for_completion` option).